### PR TITLE
Automatically compact inline CSS

### PR DESCRIPTION
**Pull request by frankiejarrett**
By compacting the inline CSS for colors/fonts I was able to remove ~430 lines from the HTML.

When `SCRIPT_DEBUG` is enabled, inline CSS will never be compacted.

### CODE REVIEW

File: inc/customizer.php [hunk 1]
```diff
@@ -233,6 +233,53 @@ public static function parse_css_rules( array $rules ) {
...
+	 * Compact CSS output.
+	 *
+	 * @link  https://github.com/GaryJones/Simple-PHP-CSS-Minification
+	 * @since NEXT
+	 *
+	 * @param  string $css
+	 *
+	 * @return string
+	 */
+	public static function compact_css( $css ) {
```
Comment (jonathanbardo) [inline inc/customizer.php]: @fjarrett 
>May not support some of the more complex syntaxes of CSS3. For that, use a proper parser and minifier.

I think we need some unit test before I feel comfortable merging that.

Comment (frankiejarrett) [reply inc/customizer.php]: @jonathanbardo I made things better with https://github.com/godaddy/wp-primer-theme/pull/150/commits/e3fba90e80d73ab16e792da321f42e9afd2c8c8c so now media queries are properly supported. But yeah I get it, I wanted to take a stab at it - and it does work. I ran Primer's `style.css` through it and tested the output with the W3C CSS validator and got back identical results for both.

**Comment by frankiejarrett** [comment]
@jonathanbardo @EvanHerman OK I made some changes that basically achieve almost the same result as before, but in a 100% safer way.

Instead of running the expanded CSS returned by the parser through a "compactor", we can just have the parser change the format of its own output based on `SCRIPT_DEBUG`. Basically solving the problem more at the source rather than after-the-fact. No regex needed.

**Comment by jonathanbardo** [comment]
@fjarrett That's way cleaner imho. Not sure the overall benefit on doing this since anyways it seems like [70% of servers](https://w3techs.com/technologies/details/ce-gzipcompression/all/all) use gzip to transfer data nowadays but I guess inclusion is always good.

**Comment by frankiejarrett** [comment]
@jonathanbardo At the very least it's the aesthetic benefit of making the HTML source cleaner. As a user if my page source could be ~430 lines shorter, I'd want that to be the default behavior.

**Review by EvanHerman — APPROVED** [review]
@fjarrett @jonathanbardo Both valid points, although until gzip usage is up to 100% I guess we'll need a fallback for those cases.

I think this approach is much cleaner - both code wise and the end results in the inspector. Nice work.

**Pull request merged**